monitor/internal/pod/monitor.go

```diff
@@ -181,6 +183,14 @@ func (m *PodMonitor) Run(ctx context.Context) error {
 		return errors.New("pod: controller did not start within 5s")
 	case <-controllerStarted:
 		m.kubeClient = mgr.GetClient()
+
+		// call ResyncWithAllPods before we return from here
+		// this will block until every pod at this point in time has been seeing at least one `Reconcile` call
+		// we do this so that we build up our internal PU cache in the policy engine,
+		// so that when we remove stale pods on startup, we don't remove them and create them again
+		if err := ResyncWithAllPods(ctx, m.kubeClient, m.resyncInfo, m.eventsCh); err != nil {
+			// not important enough to fail
```

Comment:
Should we atleast log it ? I see the`ResyncWithAllPods ` never returns any error but still to be safe.

Reply:
I guess we could make it a Debug log.